src/components/ViewedByCustomer/ViewedByCustomer.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, ComponentOptions, IComponentBindings, $$, Initialization } from 'coveo-search-ui';
+import { user } from '../../utils/icons';
+
+/**
+ * The options for the ViewedByCustomerComponent
+ */
+export interface IViewedByCustomerOptions {
+    showIcon?: boolean;
+    label?: string;
+}
+
+/**
+ * Result component indicating if a search result have been viewed by the targeted user.
+ */
+export class ViewedByCustomer extends Component {
+    /**
+     * Unique Identifier used by the Search-UI.
+     */
+    public static readonly ID = 'ViewedByCustomer';
+
+    /**
+     * Default options used by the component.
+     */
+    public static readonly options: IViewedByCustomerOptions = {
+        /**
+         * If true, will display an icon when the component is displayed itself.
+         */
+        showIcon: ComponentOptions.buildBooleanOption({ defaultValue: true }),
+        /**
+         * The label that should be displayed when the component is displayed.
+         */
+        label: ComponentOptions.buildStringOption({ defaultValue: 'Viewed by Customer' })
+    };
+
+    /**
+     * Create an instance of {@link ViewedByCustomer}.
+     * @param element Element on which to bind the component.
+     * @param options Initialization options of the component.
+     * @param bindings Bindings of the Search-UI environment.
+     */
+    public constructor(public element: HTMLElement, public options: IViewedByCustomerOptions, public bindings: IComponentBindings) {
+        super(element, ViewedByCustomer.ID, bindings)/* istanbul ignore next Issue with Istanbul and super calls*/;

```

Comment:
`/* istanbul ignore next Issue with Istanbul and super calls*/`
Is that annotation necessary?
Can we make it a general rule somewhere so we don't have to have annotations in the code?
If we can't can we make the comment line be above the line instead of on the same line?

Reply:
No we cannot, it's an issue with Istanbul that consider the semi at the end of the super as a statement (it baffles me as much as you). So, istanbul ignore next ignoring the next statement does not help if we don't put it just before the 'statement', so, here, just before the semicolon.